subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaIntegTest.groovy

```diff
@@ -849,4 +852,107 @@ $append
 
     }
 
+    def 'can publish java library using a platform with a dependencyManagement entry of type "pom" and scope "import"'() {
+        given:
+        javaLibrary(mavenRepo.module("org.test", "bom", "1.0")).hasPackaging('pom').dependencyConstraint(mavenRepo.module('org.test', 'bar', '1.1')).withModuleMetadata().publish()
+        javaLibrary(mavenRepo.module("org.test", "bar", "1.0")).withModuleMetadata().publish()
+        javaLibrary(mavenRepo.module("org.test", "bar", "1.1")).withModuleMetadata().publish()
+
+        createBuildScripts("""
+            dependencies {
+                api "org.test:bar"
+                api platform("org.test:bom:1.0")
+            }
+            publishing {
+                publications {
+                    maven(MavenPublication) {
+                        from components.javaLibraryPlatform
```

Comment:
We are publishing a _java library_, not a _java library platform_: this should be `from components.java`.

Then, you'll be able to use `javaLibrary.assertPublished()` instead of `mavenModule.assertArtifactsPublished()` below.

We should probably also verify that the `org.test:bar:` dependency is published in the POM file too.